modules/getting-started/pages/quickstart-dotnet27-aspnetcore31-visualstudio-first-query-cb65.adoc

```diff
@@ -0,0 +1,154 @@
+
+Select the API template.
+
+image::templateSelect.png[Template select in Visual Studio]
+
+Install the latest link:https://github.com/couchbaselabs/Couchbase.Extensions[Couchbase.Extensions.DependencyInjection] package from NuGet. This will install both the Couchbase .NET SDK and a Dependency Injection extension for ASP.NET Core.
+
+[Note]
+====
+This open-source project is part of Couchbase Labs, and is not commercially supported by Couchbase, Inc.
+====
+
+You can do this via the UI or the command line like so:
+
+[source,Powershell]
+----
+Install-Package Couchbase.Extensions.DependencyInjection -Version 2.0.2
+----
+
+== Step 2: ASP.NET Core Configuration
+
+There are two steps to configure your new ASP.NET Core application.
+
+First, add a "Couchbase" section to the appsettings.json file. If you've followed along with the xref:quickstart-docker-image-manual.adoc[10-minute Couchbase Docker Container Configuration], then edit your appsettings.json file to add a Couchbase section as follows:
+
+[source,JavaScript]
+----
+include::{sourcedir}/quickstart-aspnetcore31-cb65/appsettings.json[]
+----
+
+The next step is to setup Couchbase in Startup.cs.
+
+To do this, first add `services.AddCouchbase(Configuration.GetSection("Couchbase"));` to the `ConfigureServices` method. Make sure to close the connection as well. You can do this in `Configure` with `IHostApplicationLifetime`. Here is the complete Startup.cs:
+
+[source,JavaScript]
+----
+include::{sourcedir}/quickstart-aspnetcore31-cb65/Startup.cs[]
+----
+
+At this point, your ASP.NET Core application is now configured to use Couchbase.
+
+== Step 3: Create a new document
+
+Let's create two controller actions. These will correspond to two API endpoints: one will create a document and one will query a document.
+
+You can start with the `WeatherForecastController` that Visual Studio automatically generated, or you can create your own. In either case, create a constructor that has an `IBucketProvider` parameter. Use the bucketProvider to get an instance of a Couchbase bucket: `_bucket = bucketProvider.GetBucket("default")`.
+
+[source,C#,indent=0]
+----
+include::{sourcedir}/quickstart-aspnetcore31-cb65/Controllers/WeatherForecastController.cs[tag=ctor]
+----
+
+Use this bucket object to insert plain C# objects into Couchbase. They will be serialized into JSON and stored with a key: `_bucket.Insert(key, forecast);`. Create an `InsertData` POST method like so:
+
+[source,C#,indent=0]
+----
+include::{sourcedir}/quickstart-aspnetcore31-cb65/Controllers/WeatherForecastController.cs[tag=insert]
+----
+
+Compile and run your ASP.NET Core program locally with CTRL+F5. Use cURL or Postman or the HTTP tool of your choice to create a POST with JSON content to the specified route. The body of the post should look something like this:
+
+[source,JavaScript]
+----
+{
+	"Date" : "2020-04-14",
+	"TemperatureC" : 25,
+	"Summary" : "Today's weather is partly cloudy and warm."
+}
+----
+
```

Comment:
An sample curl statement would be good here, along with a sample response.

Reply:
I hesitate to use cURL because Powershell cURL is (annoyingly) not the same. However, Postman can import cURL statements, so I made a note of that and added the cURL sample.